lib/rubocop/cop/mixin/allowed_methods.rb

```diff
@@ -17,16 +17,21 @@ def allowed_method?(name)
 
       # @api public
       def allowed_methods
```

Comment:
[memo] The reason for the slow speed was due to this method.
If the following settings were used

```yaml
Style/MethodCallWithArgsParentheses:
  IgnoredMethods:
    - abort
    - after_save
    - and
    - before_save
    - delegate
    - describe
    - exec
    - exit
    - load
    - not_to
    - p
    - print
    - puts
    - raise
    - require
    - shared_context
    - shared_examples
    - star_puts
    - throw
    - to
    - warn
    - yield
```

it was returning the following results (increasing with each file)
```
["abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield", "abort", "after_save", "and", "before_save", "delegate", "describe", "exec", "exit", "load", "not_to", "p", "print", "puts", "raise", "require", "shared_context", "shared_examples", "star_puts", "throw", "to", "warn", "yield"]
```